raster-foundry/raster-foundry
# Add Ansible playbook to configure nexus service

## Overview

Adds an Ansible playbook to provision a Nexus repository manager, prepopulated with a Nexus backup from S3. Using Nexus will help speed up dependency downloads and keep us from being throttled by Maven Central.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
~- [ ] Any new SQL strings have tests~

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

- **nginx**- I configured Nexus to run behind an Nginx reverse proxy based on the instructions provided in the [nexus help documentation](https://help.sonatype.com/repomanager3/installation/run-behind-a-reverse-proxy#RunBehindaReverseProxy-nginx). Although i don't expect the office instance to have heavy usage, I _do_ expect `nginx` have a larger impact in CI where any number of builds can be running at a time. Also, being able to access the repo without a port specification seemed convenient to me.


## Testing Instructions

 * Make sure the playbook provisions the instance successfully.
```bash
$ ansible-playbook -i deployment/ansible/inventory -kK deployment/ansible/raster-foundry-nexus.yaml
```

 * After about 2 minutes (nexus takes a moment to initialize), verify that the Nexus UI is up at http://nexus.internal.azavea.com. You can login to the admin console using the credentials under `Sonatype Nexus Repository Manager` on LastPass.

* Follow the [wiki](https://github.com/azavea/raster-foundry-platform/wiki/Using-the-internal-Sonatype-Nexus-repository-manager-in-Development) instructions to configure your dev environment to use nexus, then run `scripts/update`. Check the `sbt update` output to verify that you are indeed downloading from `nexus.internal.azavea.com`

Closes azavea/raster-foundry-platform#518
